Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py

```diff
@@ -73,23 +73,61 @@ def parser(cls, parser, loggers=None):
         )
 
     @classmethod
-    def get_issue_info(cls, args, repository, **kwargs):
-        # Can give either a bug URL or a title of the new issue.
+    def get_commit_info(cls, args, repository, **kwargs):
+        commit_objects = []
+        commits_to_revert = []
+        commit_bugs = []
+        for c in args.commit:
+            try:
+                commit = repository.find(c, include_log=True)
+            except (local.Scm.Exception, ValueError) as exception:
+                sys.stderr.write('Could not find "{}"'.format(c) + '\n')
+                return None, None, None
+            commit_objects.append(commit)
+            commits_to_revert.append(commit.hash)
+            commit_bugs += [Tracker.from_string(i.link) for i in commit.issues if isinstance(Tracker.from_string(i.link).tracker, bugzilla.Tracker)]
+        return commit_objects, commits_to_revert, commit_bugs
+
+    @classmethod
+    def get_issue_info(cls, args, repository, commit_objects, commit_bugs, **kwargs):
+        # Can give either a bug URL or a title of the new issue
         # Using --revert-issue to differentiate from PR arg --issue/-i
         if not args.revert_issue:
-            prompt = 'Enter issue URL or title of new issue for the revert: '
+            prompt = 'Enter issue URL or reason for the revert: '
```

Comment:
This is great! I would also support a `--why` that auto-fills this prompt. That's probably also going to make the code a bit more clear, you're mixing the "issue" object with a "revert reason" currently.

Reply:
I think `--reason` sounds semantically better, but I agree that it's unclear. I also got rid of `--revert-issue` in favour of simply using `--issue`, which I believe now actually works due to previous changes I made.